test/e2e/node_feature_discovery.go

```diff
@@ -118,253 +118,276 @@ func cleanupCRDs(cli *nfdclient.Clientset) {
-					if nodeConf == nil {
-						e2elog.Logf("node %q has no matching rule in e2e-config, skipping...", node.Name)
-						continue
-					}
+				cleanupNode(f.ClientSet)
+				cleanupCRDs(nfdClient)
+			})
+
+			//
+			// Simple test with only the fake source enabled
+			//
+			Context("and a single worker pod with fake source enabled", func() {
+				It("it should decorate the node with the fake feature labels", func() {
 
-					// Check labels
-					e2elog.Logf("verifying labels of node %q...", node.Name)
-					for k, v := range nodeConf.ExpectedLabelValues {
-						Expect(node.Labels).To(HaveKeyWithValue(k, v))
+					fakeFeatureLabels := map[string]string{
+						nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature1": "true",
+						nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature2": "true",
+						nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature3": "true",
 					}
-					for k := range nodeConf.ExpectedLabelKeys {
-						Expect(node.Labels).To(HaveKey(k))
+
+					// Launch nfd-worker
+					By("Creating a nfd worker pod")
+					podSpecOpts := createPodSpecOpts(
+						testpod.SpecWithRestartPolicy(corev1.RestartPolicyNever),
+						testpod.SpecWithContainerImage(dockerImage),
+						testpod.SpecWithContainerExtraArgs("-oneshot", "-label-sources=fake"),
+					)
+					workerPod := testpod.NFDWorker(podSpecOpts...)
+					workerPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), workerPod, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Waiting for the nfd-worker pod to succeed")
+					Expect(e2epod.WaitForPodSuccessInNamespace(f.ClientSet, workerPod.ObjectMeta.Name, f.Namespace.Name)).NotTo(HaveOccurred())
+					workerPod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), workerPod.ObjectMeta.Name, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By(fmt.Sprintf("Making sure '%s' was decorated with the fake feature labels", workerPod.Spec.NodeName))
+					node, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), workerPod.Spec.NodeName, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					for k, v := range fakeFeatureLabels {
+						Expect(node.Labels[k]).To(Equal(v))
 					}
+
+					// Check that there are no unexpected NFD labels
 					for k := range node.Labels {
 						if strings.HasPrefix(k, nfdv1alpha1.FeatureLabelNs) {
-							if _, ok := nodeConf.ExpectedLabelValues[k]; ok {
-								continue
-							}
-							if _, ok := nodeConf.ExpectedLabelKeys[k]; ok {
-								continue
-							}
-							// Ignore if the label key was not whitelisted
-							Expect(fConf.LabelWhitelist).NotTo(HaveKey(k))
+							Expect(fakeFeatureLabels).Should(HaveKey(k))
 						}
 					}
 
-					// Check annotations
-					e2elog.Logf("verifying annotations of node %q...", node.Name)
-					for k, v := range nodeConf.ExpectedAnnotationValues {
-						Expect(node.Annotations).To(HaveKeyWithValue(k, v))
+					checkNodeFeatureObject(workerPod.Spec.NodeName)
```

Comment:
```suggestion
					checkNodeFeatureObject(node.Name)
```
We have the node already from L282, so I think better to get the name from the node itself directly.

Reply:
Thx, fixed!